src/controllers/admin.js

```diff
@@ -1,125 +1,154 @@
+    let changes = false;
+    adminProperties.forEach((property) => {
+      if (req.body[property]
+      && req.body[property].toString() !== actualAdmin[property].toString()) {
+        changes = true;
+      }
     });
-  }
-  const actualAdmin = await Admin.findById(id);
-  const adminProperties = Object.keys(actualAdmin.toObject()).slice(1, -1);
-  let changes = false;
-  adminProperties.forEach((property) => {
-    if (req.body[property]
-    && req.body[property].toString() !== actualAdmin[property].toString()) {
-      changes = true;
+    if (!changes) {
+      return res.status(200).json({
+        message: 'There were no changes',
+        data: undefined,
+        error: false,
+      });
     }
-  });
-  if (!changes) {
-    return res.status(200).json({
-      message: 'There were no changes',
-      data: undefined,
+    const found = await Admin.findOne({ dni });
+    if (found) {
+      return res.status(404).json({
+        message: `An admin with DNI ${dni} already exists`,
+        data: undefined,
+        error: true,
+      });
+    }
+    const result = await Admin.findByIdAndUpdate(
+      id,
+      {
+        firstName, lastName, dni, phone, email, city, password,
+      },
+      { new: true },
+    );
+    if (!result) {
+      return res.status(404).json({
+        message: `Admin with id ${id} was not found`,
+        data: undefined,
+        error: true,
+      });
+    }
+
+    return res.status(201).json({
+      message: `Admin ${id} updated`,
+      result,
       error: false,
     });
+  } catch (error) {
+    return res.status(400).json({
+      message: error.message,
+      data: undefined,
+      error: true,
+    });
   }
-  return Admin.findByIdAndUpdate(
-    id,
-    {
-      firstName, lastName, dni, phone, email, city, password,
-    },
-    { new: true },
-  )
-    .then((result) => {
-      if (!result) {
-        return res.status(404).json({
-          message: `Admin with id ${id} was not found`,
-          data: undefined,
-          error: true,
-        });
-      }
-      return res.status(201).json({
-        message: `Admin ${id} updated`,
-        result,
-        error: false,
-      });
-    })
-    .catch((error) => res.status(400).json(error));
 };
 
-const getAllAdmins = (req, res) => {
-  Admin.find()
-    .then((admins) => res.status(200).json({
+const getAllAdmins = async (req, res) => {
+  try {
+    const admins = await Admin.find();
+    return res.status(200).json({
       message: 'Complete admin list',
       data: admins,
       error: false,
-    }))
-    .catch((error) => res.status(400).json({
-      message: 'An error ocurred',
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: error.message,
       data: undefined,
-      error,
-    }));
+      error: true,
+    });
+  }
 };
 
-const getAdminById = (req, res) => {
+const getAdminById = async (req, res) => {
   const { id } = req.params;
-
   if (!mongoose.isValidObjectId(id)) {
```

Comment:
Maybe you should choose where to put the ID validation in every case for a cleaner code, in the update endpoint it is inside the try and in the getByID and Delete it is outside, but it works fine anyway